k4FWCore/include/k4FWCore/Consumer.h

```diff
@@ -90,6 +92,31 @@ namespace k4FWCore {
 
       // ... instead, they must implement the following operator
       virtual void operator()(const In&...) const = 0;
+
+      /**
+       * @brief    Get the input locations for a given input index
+       * @param i  The index of the input
+       * @return   A range of the input locations
+       */
+      const auto inputLocations(int i) const {
+        if (i >= sizeof...(In)) {
+          throw std::out_of_range("Called inputLocations with an index out of range");
```

Comment:
Maybe add how many would be available in the message?

Reply:
It's now there in the message